### ISSUE TITLE
error in reduce_last when using spancat with ALBERT xlarge

### ISSUE THREAD

**Issue by egumasa** [explosion/spaCy#11905]
Hello,

I was not sure if the following is an error in the source code, but I get following error when I train spacy spancat model with ALBERT xlarge. (the weird thing is that it seems working with RoBERTa model without an issue, so I was not sure this is issue for thinc).

```
============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'tagger', 'parser', 'ner',
'trainable_transformer', 'span_finder', 'spancat']
ℹ Frozen components: ['transformer', 'parser', 'tagger', 'ner']
ℹ Set annotations on update for: ['span_finder']
ℹ Initial learn rate: 0.0
E    #       LOSS TRAIN...  LOSS SPAN_...  LOSS SPANCAT  SPAN_FINDE...  SPAN_FINDE...  SPAN_FINDE...  SPANS_SC_F  SPANS_SC_P  SPANS_SC_R  SCORE 
---  ------  -------------  -------------  ------------  -------------  -------------  -------------  ----------  ----------  ----------  ------
  0       0        8344.77          18.89       1601.06           0.46           0.23          72.91        0.08        0.04       23.38    0.15
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
Epoch 1:  99%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎  | 99/100 [01:56<00:00,  1.12it/s]⚠ Aborting and saving the final best model. Encountered exception:
ValueError('all sequence lengths must be >= 0')
Traceback (most recent call last):
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/__main__.py", line 4, in <module>
    setup_cli()
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/cli/_util.py", line 71, in setup_cli
    command(prog_name=COMMAND)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/typer/main.py", line 532, in wrapper
    return callback(**use_params)  # type: ignore
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/cli/train.py", line 45, in train_cli
    train(config_path, output_path, use_gpu=use_gpu, overrides=overrides)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/cli/train.py", line 75, in train
    train_nlp(nlp, output_path, use_gpu=use_gpu, stdout=sys.stdout, stderr=sys.stderr)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/training/loop.py", line 122, in train
    raise e
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/training/loop.py", line 105, in train
    for batch, info, is_best_checkpoint in training_step_iterator:
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/training/loop.py", line 226, in train_while_improving
    score, other_scores = evaluate()
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/training/loop.py", line 281, in evaluate
    scores = nlp.evaluate(dev_corpus(nlp))
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/language.py", line 1430, in evaluate
    for eg, doc in zip(examples, docs):
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/language.py", line 1589, in pipe
    for doc in docs:
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/util.py", line 1651, in _pipe
    yield from proc.pipe(docs, **kwargs)
  File "spacy/pipeline/trainable_pipe.pyx", line 79, in pipe
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/util.py", line 1670, in raise_error
    raise e
  File "spacy/pipeline/trainable_pipe.pyx", line 75, in spacy.pipeline.trainable_pipe.TrainablePipe.pipe
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/spacy/pipeline/spancat.py", line 275, in predict
    scores = self.model.predict((docs, indices))  # type: ignore
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/model.py", line 315, in predict
    return self._func(self, X, is_train=False)[0]
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/layers/chain.py", line 55, in forward
    Y, inc_layer_grad = layer(X, is_train=is_train)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/model.py", line 291, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/layers/chain.py", line 55, in forward
    Y, inc_layer_grad = layer(X, is_train=is_train)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/model.py", line 291, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/layers/concatenate.py", line 44, in forward
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/layers/concatenate.py", line 44, in <listcomp>
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/model.py", line 291, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/layers/reduce_last.py", line 21, in forward
    Y, lasts = model.ops.reduce_last(cast(Floats2d, Xr.data), Xr.lengths)
  File "/Users/masakieguchi/opt/miniforge3/envs/spancat/lib/python3.9/site-packages/thinc/backends/ops.py", line 1204, in reduce_last
    raise ValueError(f"all sequence lengths must be >= 0")
ValueError: all sequence lengths must be >= 0
```

Here is my training environment:
```
# Name                    Version                   Build  Channel
altair                    4.2.0                    pypi_0    pypi
attrs                     22.1.0                   pypi_0    pypi
blinker                   1.5                      pypi_0    pypi
blis                      0.7.9                    pypi_0    pypi
bzip2                     1.0.8                h3422bc3_4    conda-forge
ca-certificates           2022.9.24            h4653dfc_0    conda-forge
cachetools                5.2.0                    pypi_0    pypi
catalogue                 2.0.8                    pypi_0    pypi
certifi                   2022.9.24                pypi_0    pypi
charset-normalizer        2.1.1                    pypi_0    pypi
click                     8.1.3                    pypi_0    pypi
commonmark                0.9.1                    pypi_0    pypi
confection                0.0.3                    pypi_0    pypi
cymem                     2.0.7                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
en-core-web-trf           3.4.1                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
filelock                  3.8.0                    pypi_0    pypi
gitdb                     4.0.10                   pypi_0    pypi
gitpython                 3.1.29                   pypi_0    pypi
huggingface-hub           0.8.1                    pypi_0    pypi
idna                      3.4                      pypi_0    pypi
importlib-metadata        5.1.0                    pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
jsonschema                4.17.1                   pypi_0    pypi
langcodes                 3.3.0                    pypi_0    pypi
libffi                    3.4.2                h3422bc3_5    conda-forge
libsqlite                 3.40.0               h76d750c_0    conda-forge
libzlib                   1.2.13               h03a7124_4    conda-forge
markupsafe                2.1.1                    pypi_0    pypi
murmurhash                1.0.9                    pypi_0    pypi
ncurses                   6.3                  h07bb92c_1    conda-forge
numpy                     1.23.5                   pypi_0    pypi
openssl                   3.0.7                h03a7124_0    conda-forge
packaging                 21.3                     pypi_0    pypi
pandas                    1.5.2                    pypi_0    pypi
pathy                     0.10.0                   pypi_0    pypi
pillow                    9.3.0                    pypi_0    pypi
pip                       22.3.1             pyhd8ed1ab_0    conda-forge
preshed                   3.0.8                    pypi_0    pypi
protobuf                  3.20.3                   pypi_0    pypi
pyarrow                   10.0.1                   pypi_0    pypi
pydantic                  1.10.2                   pypi_0    pypi
pydeck                    0.8.0                    pypi_0    pypi
pygments                  2.13.0                   pypi_0    pypi
pympler                   1.0.1                    pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
pyrsistent                0.19.2                   pypi_0    pypi
python                    3.9.15          hea58f1e_0_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
pytz                      2022.6                   pypi_0    pypi
pytz-deprecation-shim     0.1.0.post0              pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
readline                  8.1.2                h46ed386_0    conda-forge
regex                     2022.10.31               pypi_0    pypi
requests                  2.28.1                   pypi_0    pypi
rich                      12.6.0                   pypi_0    pypi
semver                    2.13.0                   pypi_0    pypi
setuptools                65.6.3                   pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smart-open                5.2.1                    pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
spacy                     3.4.3                    pypi_0    pypi
spacy-alignments          0.8.6                    pypi_0    pypi
spacy-experimental        0.6.0                    pypi_0    pypi
spacy-huggingface-hub     0.0.7                    pypi_0    pypi
spacy-legacy              3.0.10                   pypi_0    pypi
spacy-loggers             1.0.3                    pypi_0    pypi
spacy-transformers        1.1.8                    pypi_0    pypi
srsly                     2.4.5                    pypi_0    pypi
streamlit                 1.15.1                   pypi_0    pypi
thinc                     8.1.5                    pypi_0    pypi
thinc-apple-ops           0.1.2                    pypi_0    pypi
tk                        8.6.12               he1e0b03_0    conda-forge
tokenizers                0.12.1                   pypi_0    pypi
toml                      0.10.2                   pypi_0    pypi
toolz                     0.12.0                   pypi_0    pypi
torch                     1.11.0                   pypi_0    pypi
tornado                   6.2                      pypi_0    pypi
tqdm                      4.64.1                   pypi_0    pypi
transformers              4.21.3                   pypi_0    pypi
typer                     0.4.2                    pypi_0    pypi
typing-extensions         4.4.0                    pypi_0    pypi
tzdata                    2022.6                   pypi_0    pypi
tzlocal                   4.2                      pypi_0    pypi
urllib3                   1.26.13                  pypi_0    pypi
validators                0.20.0                   pypi_0    pypi
wasabi                    0.10.1                   pypi_0    pypi
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
zipp                      3.10.0                   pypi_0    pypi
```

I use miniforge3, on M1 max MBP.

**Comment by polm** [comment]
Can you share your config? In particular it'd help to check exactly which version of ALBERT xlarge you're using. 

Also, were you using the same version of spaCy when it worked? This looks like it might be the same as #11861. In that issue, the underlying problem is that a suggester that sometimes provides zero-length spans isn't handled correctly. In 3.3 the code will run (while probably doing the wrong thing), but will give the same error you got in 3.4.

**Comment by polm** [comment]
Sorry for not being more clear, but as mentioned [in this thread](https://github.com/explosion/spaCy/discussions/11818#discussioncomment-4260941) I think that while the code runs without error in 3.3 it may not be doing the right thing. For the models that "work", I would check that their outputs make sense. 

Setting the right learning rate can be tricky, especially with large models, but if you get it wrong you usually won't have flat zeroes as a result.

**Comment by polm** [comment]
This should be resolved by #11860 if you want to try building from master.

**Comment by egumasa** [comment]
Thank you so much for keeping me in the loop! I will be testing later this week. Thank you so much.
